flang/lib/Optimizer/Dialect/FIRType.cpp

```diff
@@ -358,7 +358,13 @@ BoxProcType::verify(llvm::function_ref<mlir::InFlightDiagnostic()> emitError,
   return emitError() << "invalid type for boxproc" << eleTy << '\n';
 }
 
-static bool canBePointerOrHeapElementType(mlir::Type eleTy) {
+static bool cannotBePointerElementType(mlir::Type eleTy) {
+  return eleTy.isa<BoxProcType, ShapeType, ShapeShiftType,
+                   SliceType, FieldType, LenType, HeapType, PointerType,
+                   ReferenceType, TypeDescType>();
+}
+
+static bool cannotBeHeapElementType(mlir::Type eleTy) {
```

Comment:
Unless you still need those, I think it might be best to revert the changes in these file.

Reply:
Done; thanks